pandas/core/internals.py

```diff
@@ -455,12 +455,11 @@ def downcast(self, dtypes=None, mgr=None):
 
         return blocks
 
-    def astype(self, dtype, copy=False, raise_on_error=True, values=None,
-               **kwargs):
-        return self._astype(dtype, copy=copy, raise_on_error=raise_on_error,
-                            values=values, **kwargs)
+    def astype(self, dtype, copy=False, errors='raise', values=None, **kwargs):
+        return self._astype(dtype, copy=copy, errors=errors, values=values,
+                            **kwargs)
 
-    def _astype(self, dtype, copy=False, raise_on_error=True, values=None,
+    def _astype(self, dtype, copy=False, errors='raise', values=None,
                 klass=None, mgr=None, **kwargs):
```

Comment:
Sure, there are two 'public' `astype(...)` methods: 

- `NDFrame.astype(...)` in `pandas/core/generic.py`
- `Block.astype(...)` in `pandas/core/internals.py`

In addition there is a 'protected' `Block._astype(...)` method in  
`pandas/core/internals.py`. Should I only put the checks in the 'public' 
methods?

Bearing in mind that the `raise_on_error` kwarg is going to be deprecated for
`DataFrame.where()` and replaced with the `errors` kwarg it would make sense to
put the code that checks the validity of the arguments in one place. Do we have 
any existing code where we put such validity checking functions/methods?

I notice that both `NDFrame` & `Block` inherit from `PandasObject` but, I'm not 
sure that this is the correct thing to do. Should I put a function in 
`pandas/core/base.py`?